### PR TITLE
Add config for jenkins container

### PR DESCRIPTION
**Pull request by ktibi**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -115,9 +115,9 @@ module "td" {
 
   name_prefix      = "${var.name_prefix}-jenkins"
   container_name   = local.container_name
-  container_image  = "cnservices/jenkins-master"
-  container_cpu    = 2048 # 2 vCPU - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
-  container_memory = 4096 # 4 GB  - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
+  container_image  = var.container_image
+  container_cpu    = var.container_cpu # 2 vCPU - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
```
Comment (jnonino) [inline main.tf:119]:
```suggestion
  container_cpu    = var.container_cpu
```

File: main.tf [hunk 2]
```diff
@@ -115,9 +115,9 @@ module "td" {
...
+  container_memory = var.container_memory # 4 GB  - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
```
Comment (jnonino) [inline main.tf:120]:
```suggestion
  container_memory = var.container_memory
```

File: variables.tf [hunk 3]
```diff
@@ -44,6 +44,32 @@ variable "enable_autoscaling" {
...
+  description = "Name of the docker image used for deploy jenkins"
+  type        = string
+  default     = "cnservices/jenkins-master"
+}
+
+#------------------------------------------------------------------------------
+# AWS ECS Container Definition Variables
+#------------------------------------------------------------------------------
+
+variable "container_memory" {
```
Comment (jnonino) [inline variables.tf:60]:
```suggestion
# https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
variable "container_memory" {
```

File: variables.tf [hunk 4]
```diff
@@ -44,6 +44,32 @@ variable "enable_autoscaling" {
...
+  type        = number
+  description = "(Optional) The amount of memory (in MiB) to allow the container to use. This is a hard limit, if the container attempts to exceed the container_memory, the container is killed. This field is optional for Fargate launch type and the total amount of container_memory of all containers in a task will need to be lower than the task memory value"
+  default     = 4096 # 4 GB
+}
+
+variable "container_cpu" {
```
Comment (jnonino) [inline variables.tf:66]:
```suggestion
# https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
variable "container_cpu" {
```

File: variables.tf [hunk 5]
```diff
@@ -44,6 +44,32 @@ variable "enable_autoscaling" {
...
+  # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs
```
Comment (jnonino) [inline variables.tf:67]:
```suggestion
```

**Review by jnonino — CHANGES_REQUESTED** [review]

**Comment by ktibi** [comment]
Hi Julian,

I change the task definition for EFS.

If we keep root_directory with your config, Fargate fail because he can found the directory in the EFS.

I don't know if we can create directly the directory in the EFS with terraform. So by default I use / .

**Comment by jnonino** [comment]
> Hi @jnonino Julian,
> 
> I change the task definition for EFS.
> 
> If we keep root_directory with your config, Fargate fail because it can't found the directory in the EFS.
> 
> I don't know if we can create directly the directory in the EFS with terraform. So by default I use / .

Hi @ktibi, from what I remember the Jenkins image creates that directory when started running but that might be changed! I don't feel too confortable with using `/` if you can find an alternative that'd be great!!!

[Working with Amazon EFS access points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html)

> Creating the root directory for an access point
If a root directory path for an access point doesn't exist on the file system, Amazon EFS automatically creates that root directory with the ownership and permissions specified.

[Terraform efs_access_point](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_access_point)

**Comment by jnonino** [comment]
> Hi @jnonino Julian,
> 
> I change the task definition for EFS.
> 
> If we keep root_directory with your config, Fargate fail because it can't found the directory in the EFS.
> 
> I don't know if we can create directly the directory in the EFS with terraform. So by default I use / .

Hi @ktibi, from what I remember the Jenkins image creates that directory when started running but that might be changed! I don't feel too confortable with using `/` if you can find an alternative that'd be great!!!

[Working with Amazon EFS access points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html)

> Creating the root directory for an access point
If a root directory path for an access point doesn't exist on the file system, Amazon EFS automatically creates that root directory with the ownership and permissions specified.

[](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_access_point)

**Review by jnonino — APPROVED** [review]

**Pull request merged**